// libfort-hermes - Tracking File I/O library.
//
// Copyright (C) 2018-2023  Universit√©e de Lausanne
//
// This file is part of libfort-hermes.
//
// libfort-hermes is free software: you can redistribute it and/or modify it
// under the terms of the GNU General Public License as published by the Free
// Software Foundation, either version 3 of the License, or (at your option) any
// later version.
//
// libfort-hermes is distributed in the hope that it will be useful, but WITHOUT
// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
// FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
// details.
//
// You should have received a copy of the GNU General Public License along with
// libfort-hermes.  If not, see <http://www.gnu.org/licenses/>.

syntax = "proto3";

option optimize_for = LITE_RUNTIME;
option go_package = "github.com/formicidae-tracker/hermes";

import "google/protobuf/timestamp.proto";
import "google/protobuf/any.proto";
import "Tag.proto";

package fort.hermes;

message FrameReadout {

	enum Error {
		NO_ERROR = 0;
		PROCESS_OVERFLOW = 1;
		PROCESS_TIMEOUT = 2;
		ILLUMINATION_ERROR = 3;
	}
	int64 timestamp = 1;
	int64 frameID = 2;
	repeated Tag tags = 3;
	google.protobuf.Timestamp time = 4;
	Error error = 5;
	string producer_uuid = 6;
	int32 quads = 7;
	int32 width = 8;
	int32 height = 9;
	google.protobuf.Timestamp cuestart = 10;
	google.protobuf.Timestamp cueend = 11;

	repeated google.protobuf.Any extensions = 12;
};
